lib/libutee/tee_api_operations.c

```diff
@@ -1100,13 +1087,15 @@ TEE_Result TEE_CipherDoFinal(TEE_OperationHandle operation,
 	} else {
 		req_dlen = srcLen;
 	}
-	if (*destLen < req_dlen) {
-		*destLen = req_dlen;
+	if (destLen)
+		tmp_dlen = *destLen;
+	if (tmp_dlen < req_dlen) {
+		if (destLen)
+			*destLen = req_dlen;
 		res = TEE_ERROR_SHORT_BUFFER;
 		goto out;
 	}
 
-	tmp_dlen = *destLen - acc_dlen;
```

Comment:
This changes `tmp_dlen` value. is this ok? `tee_buffer_update()` relies on this value.